librz/bin/p/bin_sms.c

```diff
@@ -55,14 +77,14 @@ static RzBinInfo *info(RzBinFile *bf) {
 		free(ret);
 		return NULL;
 	}
-	SMS_Header hdr = { { 0 } };
-	rz_buf_read_at(bf->buf, cb, (ut8 *)&hdr, sizeof(hdr));
-	hdr.CheckSum = rz_read_le16(&hdr.CheckSum);
+	SMS_Header *hdr = bf->o->bin_obj;
+	rz_buf_read_at(bf->buf, hdr->offset, (ut8 *)&hdr, sizeof(hdr));
+	hdr->CheckSum = rz_read_le16(&hdr->CheckSum);
```

Comment:
```suggestion
	SMS_Header *hdr = bf->o->bin_obj;
```

You already have parsed the header, why would you re-read it?